File: 02.Array-and-Hashing/JavaScript-Hashing-Solution/containsDuplicateHashSolution.js

var containsDuplicate = function(nums) {
    const counts = {};
    
    for (let num of nums) {
        if (counts[num]) {
            // If the number already exists in counts, a duplicate is found
            return true;
        }
        counts[num] = 1; // Mark the number as seen
    }
    
    // If no duplicates were found
    return false;
};

File: 01.Array-and-TwoPointer/CPP-Hashing-Solution/containsDuplicateHashSolution.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        unordered_map<int,int> counts;

        // Count occurrences of each number
        for(int x: nums) {
            counts[x]++;
        }

        // Check if any number occurs more than once
        for(auto x: counts) {
            if(x.second >= 2) return true; //  condition to check duplicates
        }
        return false; // Return false if no duplicates found
    }
};